### PR TITLE
Use eventfilterinformer

### PR DESCRIPTION
**Pull request by qiujian16**

### CODE REVIEW

File: pkg/spoke/hubclientcert/controller.go [hunk 1]
```diff
@@ -103,34 +103,30 @@ func NewClientCertForHubController(
...
-			return irrelevantSecretKey
-		}, spokeSecretInformer.Informer()).
+			},
+			func(obj interface{}) bool {
+				accessor, err := meta.Accessor(obj)
+				if err != nil {
+					return false
+				}
+				// only enqueue when hub kubeconfig secret is changed
+				if accessor.GetNamespace() == hubKubeconfigSecretNamespace || accessor.GetName() == kubeconfigSecretName {
```
Comment (skeeey) [inline pkg/spoke/hubclientcert/controller.go]: I think this should be `&&`, right?

File: pkg/spoke/hubclientcert/secret_controller.go [hunk 2]
```diff
@@ -47,27 +47,28 @@ func NewHubKubeconfigSecretController(
...
+			func(obj interface{}) bool {
 				accessor, err := meta.Accessor(obj)
 				if err != nil {
-					return ""
+					return false
 				}
-				if accessor.GetNamespace() == hubKubeconfigSecretNamespace && accessor.GetName() == hubKubeconfigSecretName {
-					return accessor.GetName()
+				// only enqueue when hub kubeconfig secret is changed
+				if accessor.GetNamespace() == hubKubeconfigSecretNamespace || accessor.GetName() == hubKubeconfigSecretName {
```
Comment (skeeey) [inline pkg/spoke/hubclientcert/secret_controller.go]: I think this should be `&&`, right?

File: pkg/spoke/hubclientcert/controller.go [hunk 3]
```diff
@@ -103,34 +103,30 @@ func NewClientCertForHubController(
...
-			if accessor.GetNamespace() == hubKubeconfigSecretNamespace && accessor.GetName() == kubeconfigSecretName {
+		WithFilteredEventsInformersQueueKeyFunc(
+			func(obj runtime.Object) string {
+				accessor, _ := meta.Accessor(obj)
 				return accessor.GetName()
-			}
-			return irrelevantSecretKey
-		}, spokeSecretInformer.Informer()).
+			},
+			func(obj interface{}) bool {
```
Comment (deads2k) [inline pkg/spoke/hubclientcert/controller.go:111]: I'm open to pushing this into a function in library-go for, `filterByInstance(ns, name) func` or some such.

**Pull request merged**